google/cloud/bigtable/data/_async/client.py

```diff
@@ -524,20 +538,37 @@ async def read_rows(
         query: ReadRowsQuery | dict[str, Any],
         *,
         operation_timeout: float | None = None,
-        per_request_timeout: float | None = None,
+        attempt_timeout: float | None = None,
     ) -> list[Row]:
         """
-        Helper function that returns a full list instead of a generator
+        Read a set of rows from the table, based on the specified query.
+        Retruns results as a list of Row objects when the request is complete.
+        For streamed results, use read_rows_stream.
 
-        See read_rows_stream
+        Failed requests within operation_timeout will be retried.
 
+        Args:
+            - query: contains details about which rows to return
+            - operation_timeout: the time budget for the entire operation, in seconds.
+                 Failed requests will be retried within the budget.
+                 If None, defaults to the Table's default_read_rows_operation_timeout
+            - attempt_timeout: the time budget for an individual network request, in seconds.
+                If it takes longer than this time to complete, the request will be cancelled with
+                a DeadlineExceeded exception, and a retry will be attempted.
+                If None, defaults to the Table's default_read_rows_attempt_timeout,
+                or the operation_timeout if that is also None.
         Returns:
-            - a list of the rows returned by the query
+            - a list of Rows returned by the query
+        Raises:
+            - DeadlineExceeded: raised after operation timeout
+                will be chained with a RetryExceptionGroup containing GoogleAPIError exceptions
+                from any retries that failed
+            - GoogleAPIError: raised if the request encounters an unrecoverable error
```

Comment:
Nit: I think we should rename this to GoogleApiError. I couldn't find the style guide on this, but this seems to be the convention for some other classes, example: https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-api-gateway/google/cloud/apigateway_v1/types/apigateway.py#L53

Reply:
That does look better, but I'm referring to [this class](https://github.com/googleapis/python-api-core/blob/0acda8c96b23af2238ae0f4536e4fbb98bc7d787/google/api_core/exceptions.py#L59) in `api_core` here, which is used across out libraries. So we can't rename it unfortunately